presto-spi/src/main/java/io/prestosql/spi/block/LongArrayBlockEncoding.java

```diff
@@ -53,9 +54,14 @@ public Block readBlock(BlockEncodingSerde blockEncodingSerde, SliceInput sliceIn
         boolean[] valueIsNull = decodeNullBits(sliceInput, positionCount).orElse(null);
 
         long[] values = new long[positionCount];
-        for (int position = 0; position < positionCount; position++) {
-            if (valueIsNull == null || !valueIsNull[position]) {
-                values[position] = sliceInput.readLong();
+        if (valueIsNull == null) {
+            sliceInput.readBytes(Slices.wrappedLongArray(values), 0, Long.BYTES * positionCount);
```

Comment:
You might need to handle the case when `(positionCount * Long.BYTES) > Integer.MAX_VALUE`

Reply:
As an aside, maybe we should consider implementing`SliceInput#readLongs(long[], int, int)` and similar methods for other primitive types?